liquidity/liquidity.go

```diff
@@ -641,7 +693,195 @@ func (m *Manager) dispatchBestEasyAutoloopSwap(ctx context.Context) error {
 
 	suggestion, err := builder.buildSwap(
 		ctx, channel.PubKeyBytes, outgoing, swapAmt, easyParams,
-		nil,
+	)
+	if err != nil {
+		return err
+	}
+
+	var swp loop.OutRequest
+	if t, ok := suggestion.(*loopOutSwapSuggestion); ok {
+		swp = t.OutRequest
+	} else {
+		return fmt.Errorf("unexpected swap suggestion type: %T", t)
+	}
+
+	// Dispatch a sticky loop out.
+	go m.dispatchStickyLoopOut(
+		ctx, swp, defaultAmountBackoffRetry, defaultAmountBackoff,
+	)
+
+	return nil
+}
+
+// dispatchBestAssetEasyAutoloopSwap tries to dispatch a swap to bring the total
+// local balance back to the target for the given asset.
+func (m *Manager) dispatchBestAssetEasyAutoloopSwap(ctx context.Context,
+	assetID string, localTarget uint64) error {
+
+	if assetID == "" && len(assetID) != 32 {
+		return fmt.Errorf("invalid asset id: %v", assetID)
+	}
+
+	// Retrieve existing swaps.
+	loopOut, err := m.cfg.ListLoopOut(ctx)
+	if err != nil {
+		return err
+	}
+
+	loopIn, err := m.cfg.ListLoopIn(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Get a summary of our existing swaps so that we can check our autoloop
+	// budget.
+	summary := m.checkExistingAutoLoops(ctx, loopOut, loopIn)
+
+	err = m.checkSummaryBudget(summary)
+	if err != nil {
+		return err
+	}
+
+	_, err = m.checkSummaryInflight(summary)
+	if err != nil {
+		return err
+	}
+
+	// Get all channels in order to calculate current total local balance.
+	channels, err := m.cfg.Lnd.Client.ListChannels(ctx, false, false)
+	if err != nil {
+		return err
+	}
+
+	// If we are running a custom asset, we'll need to get a random asset
+	// peer pubkey in order to rfq the asset price.
+	var assetPeerPubkey []byte
+
+	localTotal := uint64(0)
+	for _, channel := range channels {
+		// We'll only interested in custom asset channels.
+		if !channelIsCustom(channel) {
+			continue
+		}
+
+		assetData := getCustomAssetData(channel, assetID)
+		if assetData == nil {
+			continue
+		}
+
+		// We'll overwrite the channel local balance to be
+		// the custom asset balance. This allows us to make
+		// use of existing logic.
+		channel.LocalBalance = btcutil.Amount(
+			assetData.LocalBalance,
+		)
+
+		assetPeerPubkey = channel.PubKeyBytes[:]
+
+		localTotal += assetData.LocalBalance
+
+		// We'll overwrite the channel local balance in order to
+		// reuse channel selection logic.
+		channel.LocalBalance = btcutil.Amount(localTotal)
```

Comment:
we are setting the `LocalBalance` already above, is it correct to overwrite it here?